templates/custom-indexer-template/src/components/providers/WalletProvider.tsx

```diff
@@ -0,0 +1,32 @@
+"use client";
```

Comment:
can we use the same Wallet components structure as in the other nextjs template? https://github.com/aptos-labs/create-aptos-dapp/tree/main/templates/nextjs-boilerplate-template/src/components

1. place all files in the components folder
2. define optInWallets